crontab/finish_smoothreading.php

```diff
@@ -0,0 +1,67 @@
+<?php
+$relPath="./../pinc/";
+include_once($relPath.'base.inc');
+include_once($relPath.'misc.inc');
+include_once($relPath.'project_states.inc');
+include_once($relPath.'project_events.inc');
+include_once($relPath.'user_project_info.inc');
+include_once($relPath.'Project.inc');
+
+// check that caller is localhost or bail
+if(!requester_is_localhost())
+    die("You are not authorized to perform this request.");
+
+header('Content-type: text/plain');
+
+$dry_run = array_get( $_GET, 'dry_run', '' );
+if ($dry_run)
+{
+    echo "This is a dry run.\n";
+}
+
+// Interval relative to current time
+//   Select all projects whose smooth reading deadline is within this interval.
+$from =  -60 * 60;
+$to   =   60 * 60;
+
+$result = mysqli_query(DPDatabase::get_connection(), "
+    SELECT *
+    FROM projects
+    WHERE
+        smoothread_deadline >= (UNIX_TIMESTAMP() + $from) 
+        AND smoothread_deadline <= (UNIX_TIMESTAMP() + $to) 
+        AND state = '".PROJ_POST_FIRST_CHECKED_OUT."'
+") or die(mysqli_error(DPDatabase::get_connection()));
+
+echo "Checking ", mysqli_num_rows($result), " projects...\n";
+
+while ( $project = mysqli_fetch_assoc($result) )
```

Comment:
I thought `Project.inc` does not query the database if it receives an array?

```php
        elseif ( is_array($arg) )
        {
            // $arg is assumed to be an associative array, such
            // as would be returned by mysqli_fetch_assoc().
            $row = $arg;
        }
```

From [here](https://github.com/DistributedProofreaders/dproofreaders/blob/master/pinc/Project.inc#L32-L37). Then isn't it more efficient to grab all the `Project`s with *one* query, rather than only selecting the `projectid` and than querying each `Project` individually?

Reply:
Sorry, you are correct and I missed that.